dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java

```diff
@@ -210,7 +211,13 @@ public Map<URL, Set<NotifyListener>> getOverrideListeners() {
     }
 
     private void register(Registry registry, URL registeredProviderUrl) {
-        registry.register(registeredProviderUrl);
+        ApplicationDeployer deployer = registeredProviderUrl.getOrDefaultApplicationModel().getDeployer();
+        try {
+            deployer.increaseServiceRefreshCount();
+            registry.register(registeredProviderUrl);
+        } finally {
+            deployer.decreaseServiceRefreshCount();
```

Comment:
The refresh starting state should be marked at the beginning of a batch service update, and being marked as finished when batch updating finished. Currently, the state only is the state of one service, so ephemeral revisions might still happen.